components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1683,6 +1684,32 @@ void AdsServiceImpl::OnPrefsChanged(
     } else {
       Stop();
     }
+
+    // Record P3A.
+    using brave_rewards::AdsP3AState;
+    const bool rewards_enabled =
+        profile_->GetPrefs()->GetBoolean(
+            brave_rewards::prefs::kBraveRewardsEnabled);
+    if (pref == prefs::kEnabled) {
+      if (profile_->GetPrefs()->GetBoolean(prefs::kEnabled)) {
+        brave_rewards::RecordAdsState(AdsP3AState::kAdsEnabled);
+        profile_->GetPrefs()->SetBoolean(prefs::kBraveAdsWereDisabled, false);
+      } else {
+        // Apparently, the pref was disabled.
+        brave_rewards::RecordAdsState(
+            rewards_enabled ? AdsP3AState::kAdsEnabledThenDisabledRewardsOn :
+                              AdsP3AState::kAdsEnabledThenDisabledRewardsOff);
+        profile_->GetPrefs()->SetBoolean(prefs::kBraveAdsWereDisabled, true);
+      }
+    } else {
+      // Rewards pref was changed
```

Comment:
this is not a safe assumption. There are other prefs that trigger this method and more could be added in the future

Reply:
fixed